src/java/main/org/apache/zookeeper/server/EphemeralType.java

```diff
@@ -40,19 +39,21 @@
     TTL;
 
     public static final long CONTAINER_EPHEMERAL_OWNER = Long.MIN_VALUE;
-    public static final long MAX_TTL = 0x0fffffffffffffffL;
-    public static final long TTL_MASK = 0x8000000000000000L;
+    public static final long MAX_TTL = 0x00ffffffffffffffL;
+    public static final long TTL_MASK = 0xff00000000000000L;
+    public static final long MAX_TTL_SERVER_ID = 0xfe;  // 254
 
     public static EphemeralType get(long ephemeralOwner) {
         if (ephemeralOwner == CONTAINER_EPHEMERAL_OWNER) {
             return CONTAINER;
         }
-        if (ephemeralOwner < 0) {
+        if ((ephemeralOwner & TTL_MASK) == TTL_MASK) {
```

Comment:
I think this is broken. If, for example, org.apache.zookeeper.server.DataTree#deleteNode calls this method with TTL turned off, but using a server id of 255 it will end up calling this code

            } else if (ephemeralType == EphemeralType.TTL) {
                ttls.remove(path);

and the node will not be removed properly in deleteNode.

Reply:
In particular you need to add a bunch more testing to ensure this case is working properly.